docs/content/docs/concepts/stateful-stream-processing.md

```diff
@@ -342,26 +342,3 @@ give *exactly once* guarantees even in *at least once* mode.
 {{< /hint >}}
 
 {{< top >}}
-
-## State and Fault Tolerance in Batch Programs
-
-Flink executes [batch programs]({{< ref "docs/dev/dataset/overview" >}}) as a special case of
-streaming programs, where the streams are bounded (finite number of elements).
-A *DataSet* is treated internally as a stream of data. The concepts above thus
-apply to batch programs in the same way as well as they apply to streaming
-programs, with minor exceptions:
-
-  - [Fault tolerance for batch programs]({{< ref "docs/ops/state/task_failure_recovery" >}})
-    does not use checkpointing.  Recovery happens by fully replaying the
-    streams.  That is possible, because inputs are bounded. This pushes the
-    cost more towards the recovery, but makes the regular processing cheaper,
-    because it avoids checkpoints.
-
-  - Stateful operations in the DataSet API use simplified in-memory/out-of-core
-    data structures, rather than key/value indexes.
-
-  - The DataSet API introduces special synchronized (superstep-based)
-    iterations, which are only possible on bounded streams. For details, check
-    out the [iteration docs]({{< ref "docs/dev/dataset/iterations" >}}).
```

Comment:
I don't think these should all be removed. These contents also applies to DataStream jobs in batch execution mode.

I'd suggest:
1. Link `batch programs` to DataStream Batch Execution mode: `docs/dev/datastream/execution_mode`
2. Remove "A *DataSet* is treated internally as a stream of data."
3. Change "Stateful operations in the DataSet API" to "State backend in batch execution mode".
4. Remove "The DataSet API introduces special synchronized (superstep-based) iterations, which are only possible on bounded streams. For details, check out the [iteration docs]({{< ref "docs/dev/dataset/iterations" >}}).".

Reply:
sure. 
This is a good point.
I will search the others to see if any other same pattern, which could be executed in BATCH mode.